### ISSUE TITLE
ccxtpro.kraken.Watch_order_book only follows first 20 symbols

### ISSUE THREAD

**Issue by 1key** [ccxt/ccxt#6588]
- OS: Win10
- Programming Language version: Python 3.8
- CCXT version: PRO 0.0.60
- Exchange: Kraken
- Method: watch_order_book()

With kraken, only the first 20 markets will be watched. This is not the case for other exchanges like Poloniex.

```
import asyncio
import ccxtpro

sym = []

def checkSym(symbol):
    if not symbol in sym:
        sym.append(symbol)
        print(symbol)
        print(str(len(sym)))

async def loop(exchange, symbol):
    print("symbol: "+str(symbol))
    if not symbol.endswith('.d'):
        while True:
            try:
                orderbook = await exchange.watch_order_book(symbol)
                now = exchange.milliseconds()
                checkSym(symbol)
            except Exception as e:
                print(str(e))

async def main():
    exchange = ccxtpro.kraken({'enableRateLimit': True})
    await exchange.load_markets()
    sb = exchange.markets
    for symbol in sb:
        print(symbol)
    await asyncio.gather(*[loop(exchange, symbol) for symbol in sb])
    await exchange.close()

if __name__ == '__main__':
    asyncio.get_event_loop().run_until_complete(main())
```

Result:

```
ADA/ETH
1
ADA/EUR
2
<...>
BCH/BTC
19
DAI/EUR
20
```

**Comment by kroitor** [comment]
This is coming from Kraken limitations itself, but, if you can enable the verbose mode like icytrader did [here](https://github.com/ccxt/ccxt/issues/6558#issuecomment-596072724) and post your output in a gist or text file here, it will help resolve the issue quicker (add `exchange.verbose = True` right after the call to `load_markets()`).

**Comment by 1key** [comment]
...will resolve even quicker... :-)

The verbose text is a bit too long because of all websocket response from the orderbook. But I could find this message starting from nr 21:
`2020-03-08T10:56:15.401Z message {"event":"error","errorMessage":"Exceeded msg rate"}`

It can be fixed with an `await asyncio.sleep(0.5)` inside a loop triggering each WatchOrderBook().
So, we can work around this just fine, but maybe you could somehow add this function in regards to 'enableratelimit': True?

**Comment by kroitor** [comment]
@1key yep, we will add more logic to ws rate limit handling, and it will throttle them automatically, will get back to you on this one.

**Comment by kroitor** [comment]
@1key btw, while we're fixing it, a quick question to discuss –  in general the exchanges will not like it, if you're trying to subscribe to all symbols from one location, since that is a lot of data to send and handle – and it will be very hard if there's a lot of active pairs. Is there a particular reason why you would want to listen to all orderbooks? Because that is often not the case with real-world real-time trading systems which trade specifically configured symbols. Can you elaborate a bit on your use case?

**Comment by 1key** [comment]
@kroitor Good question.
I've got a theory that seems to work pretty well according to backtesting. But I'm still to affraid to go live 'in large'. So I've been playing with CCXT and mainly waiting for CCXTpro for half a yera. At the moment I still call this 'playing around' and testing if my formulas work out in realtime/reallive. 
I'm a programmer by eduction, but don't do much programming in my work. So this can be seen as hobby with a nice potential profit. :-)

**Comment by 1key** [comment]
> ...will resolve even quicker... :-)
> 
> The verbose text is a bit too long because of all websocket response from the orderbook. But I could find this message starting from nr 21:
> `2020-03-08T10:56:15.401Z message {"event":"error","errorMessage":"Exceeded msg rate"}`
> 
> It can be fixed with an `await asyncio.sleep(0.5)` inside a loop triggering each WatchOrderBook().
> So, we can work around this just fine, but maybe you could somehow add this function in regards to 'enableratelimit': True?

Same goes to gateIO!

Verbose error:
(already after 3(!) watch_order_books())
`2020-03-08T12:25:52.111Z message {"error": {"code": 8, "message": "requests too quickly"}, "result": null, "id": 0}
`
Can be fixed with sleep() also.

**Comment by kroitor** [comment]
@1key yep, we will add a generalized solution that will work for all of them. Thx again! Will post updates as soon as we have them.

**Comment by kroitor** [comment]
The fix for Python and JS will be available within a couple of hours after we finish some testing, the PHP fix is also on the way. Thx for your patience!

**Comment by kroitor** [comment]
@1key this is fixed in JS and Python now in CCXT Pro 0.0.61+.

The PHP fix is on the way as well.

You can try this example

- https://github.com/ccxt/ccxt/blob/master/examples/ccxt.pro/py/watch-all-symbols.py

```Python
# -*- coding: utf-8 -*-

import asyncio
import ccxtpro


async def loop(exchange, symbol, n):
    i = 0
    while True:
        try:
            orderbook = await exchange.watch_order_book(symbol)
            # print every 100th bidask to avoid wasting CPU cycles on printing
            if not i % 100:
                # i = how many updates there were in total
                # n = the number of the pair to count subscriptions
                now = exchange.milliseconds()
                print(exchange.iso8601(now), n, symbol, i, orderbook['asks'][0], orderbook['bids'][0])
            i += 1
        except Exception as e:
            print(str(e))
            # raise e  # uncomment to break all loops in case of an error in any one of them
            # break  # you can also break just this one loop if it fails


async def main():
    exchange = ccxtpro.kraken({'enableRateLimit': True})
    await exchange.load_markets()
    markets = list(exchange.markets.values())
    symbols = [market['symbol'] for market in markets if not market['darkpool']]
    await asyncio.gather(*[loop(exchange, symbol, n) for n, symbol in enumerate(symbols)])
    await exchange.close()


if __name__ == '__main__':
    asyncio.get_event_loop().run_until_complete(main())
```

It outputs two numbers:
- `n` is the number of the pair – watch it grow over 20...
- `i` is the number of updates received by that pair

It will output every 100th `i`, otherwise your screen and CPU will burn due to high load. In fact listening all symbols this way in a centralized script is not recommended due to architectural considerations, very few exchanges will tolerate than amount of data, and you probably won't be able to process it with just one host as quickly as the exchanges' infrastructures do. 

Let me know if that does not help ) It will be great if you report back whether the issue is solved on your side or not.
